python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py

```diff
@@ -0,0 +1,188 @@
+            self._set_span_attribute(
+                span, SpanAttributes.OUTPUT_VALUE, result.choices[0].message.content
+            )
+        elif isinstance(result, litellm.EmbeddingResponse):
+            if len(result.data) > 0:
+                first_embedding = result.data[0]
+                self._set_span_attribute(
+                    span,
+                    EmbeddingAttributes.EMBEDDING_VECTOR,
+                    json.dumps(first_embedding.get("embedding", [])),
+                )
+        elif isinstance(result, litellm.ImageResponse):
+            if len(result.data) > 0:
+                self._set_span_attribute(span, ImageAttributes.IMAGE_URL, result.data[0]["url"])
+                self._set_span_attribute(span, SpanAttributes.OUTPUT_VALUE, result.data[0]["url"])
+        if hasattr(result, "usage"):
+            self._set_span_attribute(
+                span, SpanAttributes.LLM_TOKEN_COUNT_PROMPT, result.usage["prompt_tokens"]
+            )
+            self._set_span_attribute(
+                span, SpanAttributes.LLM_TOKEN_COUNT_COMPLETION, result.usage["completion_tokens"]
+            )
+            self._set_span_attribute(
+                span, SpanAttributes.LLM_TOKEN_COUNT_TOTAL, result.usage["total_tokens"]
+            )
+
+    def _instrument(self, tracer_provider: Optional[TracerProvider] = None) -> None:
+        if tracer_provider:
+            self.tracer_provider = tracer_provider
+            trace.set_tracer_provider(tracer_provider)
+        self.tracer = trace.get_tracer(__name__)
+
+        functions_to_instrument = {
+            "completion": "completion",
+            "acompletion": "acompletion",
+            "completion_with_retries": "completion_with_retries",
+            # Bug report filed on GitHub for acompletion_with_retries: https://github.com/BerriAI/litellm/issues/4908
+            # 'acompletion_with_retries': 'acompletion_with_retries',
+            "embedding": "embedding",
+            "aembedding": "aembedding",
+            "image_generation": "image_generation",
+            "aimage_generation": "aimage_generation",
+        }
+
+        for func_name, span_name in functions_to_instrument.items():
+            if hasattr(litellm, func_name):
+                original_func = getattr(litellm, func_name)
+                LiteLLMInstrumentor.original_litellm_funcs[func_name] = (
+                    original_func  # Add original liteLLM function to dictionary
+                )
+                setattr(litellm, func_name, self._common_wrapper(original_func, span_name))
+
+    def _uninstrument(self, **kwargs: Any) -> None:
+        for func_name, original_func in LiteLLMInstrumentor.original_litellm_funcs.items():
+            setattr(litellm, func_name, original_func)
+        LiteLLMInstrumentor.original_litellm_funcs.clear()
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return ["litellm"]
+
+    def instrument(self, **kwargs: Any) -> None:
```

Comment:
you shouldn't need to implement these, if you don't define this method on a subclass it'll fall back to `super().instrument` anyway

Reply:
got it, removed instrument() and uninstrument()